### PR TITLE
Badge icon on Linux gone missing

### PR DESCRIPTION
**Pull request by gregor**

### CODE REVIEW

File: electron/js/menu/tray.js [hunk 1]
```diff
@@ -76,9 +76,16 @@ module.exports = {
...
         app.setBadgeCount(parseInt(count, 10));
-      }
+        if (count) {
+          this.useBadgeIcon();
+        } else {
+          this.useDefaultIcon();
+        }
+      } else {
+      app.setBadgeCount(parseInt(count, 10));
+    }
```
Comment (lipis) [inline electron/js/menu/tray.js]: indentation..

Comment (herrmannplatz) [inline electron/js/menu/tray.js]: var count = parseInt(counter[0], 10) || 0;

parse count in the beginning

**Comment by gregor** [comment]
Works great on Windows

File: electron/js/menu/tray.js [hunk 2]
```diff
@@ -61,25 +61,16 @@ module.exports = {
...
-          if (!win.isFocused() && count > lastUnreadCount) {
-            win.flashFrame(true);
-          }
-        } else {
-          win.flashFrame(false);
-          win.setOverlayIcon(null, '');
-          this.useDefaultIcon();
-        }
+      let counter = (/\(([0-9]+)\)/).exec(win.getTitle() || (win.webContents ? win.webContents.getTitle() : ''));
+      let count = parseInt(counter ? counter[1] : '0', 10);
```
Comment (gregor) [inline electron/js/menu/tray.js]: let count = parseInt(counter ? counter[1] : 0, 10);

Comment (gregor) [inline electron/js/menu/tray.js]: No need to pass the else as a string

Comment (lipis) [inline electron/js/menu/tray.js]: true :)

**Comment by lipis** [comment]
Looks like it works on my Ubuntu as well.. :) Maybe we could merge it and tweak it later if needed.

**Comment by alien2003** [comment]
Stlill not fixed, omg...

**Pull request merged**